framework/src/modules/chain/modules/blocks/chain.js

```diff
@@ -836,7 +836,7 @@ Chain.prototype.deleteLastBlock = function(cb) {
 			},
 			updateSystemHeaders(seriesCb) {
 				// Update our own headers: broadhash and height
-				components.system.update(seriesCb);
+				return components.system.update().then(() => setImmediate(seriesCb));
```

Comment:
I think this can be written as follows:
```
await components.system.update();
return setImmediate(seriesCb));
```

Reply:
We cannot use `await` because the function is not `async`.